src/mongocrypt-ctx.c

```diff
@@ -294,14 +293,20 @@ mongocrypt_ctx_t *mongocrypt_ctx_new(mongocrypt_t *crypt) {
         CLIENT_ERR("cannot create context from uninitialized crypt");
         return NULL;
     }
-    ctx_size = sizeof(_mongocrypt_ctx_encrypt_t);
-    if (sizeof(_mongocrypt_ctx_decrypt_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_decrypt_t);
-    }
-    if (sizeof(_mongocrypt_ctx_datakey_t) > ctx_size) {
-        ctx_size = sizeof(_mongocrypt_ctx_datakey_t);
-    }
-    ctx = bson_malloc0(ctx_size);
+    // Allocate with maximum size and alignment of any of the possible context types.
+    size_t ctx_size = 0;
+    ctx_size = BSON_MAX(ctx_size, sizeof(_mongocrypt_ctx_encrypt_t));
+    ctx_size = BSON_MAX(ctx_size, sizeof(_mongocrypt_ctx_decrypt_t));
+    ctx_size = BSON_MAX(ctx_size, sizeof(_mongocrypt_ctx_datakey_t));
+    ctx_size = BSON_MAX(ctx_size, sizeof(_mongocrypt_ctx_rewrap_many_datakey_t));
+    ctx_size = BSON_MAX(ctx_size, sizeof(_mongocrypt_ctx_compact_t));
+    size_t ctx_alignment = 0;
+    ctx_alignment = BSON_MAX(ctx_alignment, BSON_ALIGNOF(_mongocrypt_ctx_encrypt_t));
+    ctx_alignment = BSON_MAX(ctx_alignment, BSON_ALIGNOF(_mongocrypt_ctx_decrypt_t));
+    ctx_alignment = BSON_MAX(ctx_alignment, BSON_ALIGNOF(_mongocrypt_ctx_datakey_t));
+    ctx_alignment = BSON_MAX(ctx_alignment, BSON_ALIGNOF(_mongocrypt_ctx_rewrap_many_datakey_t));
+    ctx_alignment = BSON_MAX(ctx_alignment, BSON_ALIGNOF(_mongocrypt_ctx_compact_t));
```

Comment:
```suggestion

    // Allocate enough memory for any possible context type.
    static const size_t ctx_size = BSON_MAX(
        sizeof(_mongocrypt_ctx_encrypt_t),
        BSON_MAX(sizeof(_mongocrypt_ctx_decrypt_t),
                 BSON_MAX(sizeof(_mongocrypt_ctx_datakey_t),
                          BSON_MAX(sizeof(_mongocrypt_ctx_rewrap_many_datakey_t), sizeof(_mongocrypt_ctx_compact_t)))));

    // Use an alignment that is enough for any possible context type.
    static const size_t ctx_alignment =
        BSON_MAX(BSON_ALIGNOF(_mongocrypt_ctx_encrypt_t),
                 BSON_MAX(BSON_ALIGNOF(_mongocrypt_ctx_decrypt_t),
                          BSON_MAX(BSON_ALIGNOF(_mongocrypt_ctx_datakey_t),
                                   BSON_MAX(BSON_ALIGNOF(_mongocrypt_ctx_rewrap_many_datakey_t),
                                            BSON_ALIGNOF(_mongocrypt_ctx_compact_t)))));

```
Suggest using a constant variable for these values, as they are known at compile-time, so preferably the result should also be computed at compile-time. The nested calls to `BSON_MAX` is messy but imo works well enough in this circumstance.